src/live/src/user-to-user/stream.js

```diff
@@ -0,0 +1,249 @@
+import Client from '../../../client';
+import { getLiveService } from './live-service';
+import { KinveyRequest, RequestMethod, StatusCode } from '../../../request';
+import { StreamACL } from './stream-acl';
+
+/**
+ * @typedef MessageReceiver
+ * @property {Function} status
+ * @property {Function} message
+ * @property {Function} presence
+ */
+
+/**
+ * A Stream, created in the backend
+ * @class Stream
+ */
+export class Stream {
+  /** @type {string} */
+  name;
+  /** @type {Client} @private */
+  _client = Client.sharedInstance();
+  /** @private */
+  _liveService = getLiveService(this._client);
+  /** @private */
+  _subscribeChannels = {};
+  /** @private */
+  _publishChannels = {};
+
+  /**
+   * @param {string} name
+   */
+  constructor(name) {
+    this.name = name;
+  }
+
+  getSubstreams() {
+    return this._makeStreamRequest('_substreams', RequestMethod.GET);
+  }
+
+  /**
+   * Thist method sets (overwrites) the ACL for the given user's message stream
+   * @param {string} userId The ID of the user whose stream ACL is being set
+   * @param {StreamACL} acl The ACL object to be set
+   * @returns {Promise}
+   */
+  setACL(userId, acl) {
+    const requestBody = (acl instanceof StreamACL) ? acl.toPlainObject() : acl;
+    return this._makeStreamRequest(userId, RequestMethod.PUT, requestBody);
+  }
+
+  // Feed comm
+
+  /**
+   * Subscribes the active user to the specified user's channel
+   * @param {string} userId
+   * @param {MessageReceiver} receiver
+   */
+  follow(userId, receiver) {
+    return this._subscribe(userId, receiver);
+  }
+
+  /**
+   * Unsubscribes the active user from the specified user's channel
+   * @param {string} userId
+   * @returns {Promise}
+   */
+  unfollow(userId) {
+    return this._unsubscribe(userId);
+  }
+
+  /**
+   * Posts a message to the active user's channel
+   * @param {Object} message
+   * @returns {Promise}
+   */
+  post(message) {
+    const userId = this._client.activeUser._id;
+    return this._publish(userId, message);
+  }
+
+  // Directed comm
+
+  /**
+   * Listens for messages sent to the active user
+   * @param {MessageReceiver} receiver
+   */
+  listen(receiver) {
+    const userId = this._client.activeUser._id;
+    return this._subscribe(userId, receiver);
+  }
+
+  /**
+   * Stops listening for messages sent to the active user
+   */
+  stopListening() {
```

Comment:
Is this the same as calling `stream.unfollow('<userId>')`? The comment for the functions seem to indicate that they do the same thing.